code/obj/machinery/arc_electroplater.dm

```diff
@@ -45,8 +45,31 @@ TYPEINFO(/obj/machinery/arc_electroplater)
 		SubscribeToProcess()
 		return 1
 
-	attackby(obj/item/W, mob/user)
-		if (isghostdrone(user) || isAI(user))
+	// It is time for borgs to get in on this hot electroplating action.
+	MouseDrop_T(var/obj/item/W, var/mob/user)
+		src.PlaterInteract(W, user)
+
+	attackby(var/obj/item/W, var/mob/user)
+		src.PlaterInteract(W, user)
+
+	// See if I can piece together how to make this fly.
+	proc/PlaterInteract(var/obj/item/W, var/mob/user)
+		// Please don't drag the nano-crucible into the plater. Or any other machine or mob for that matter.
+		if (!istype(W, /obj/item))
+			boutput(user, SPAN_ALERT("Please do not attempt to plate machinery or entities."))
+			return
+		// Theres a suicide button for this.
+		if (W == user)
+			boutput(user, SPAN_ALERT("Please do not drag yourself into the Plater."))
+			return
+		// Do not attempt to plate objects from a distance.
+		if (BOUNDS_DIST(user, src) > 0)
+			return
+		// Do not attempt to plate objects at a distance.
+		if (BOUNDS_DIST(W, src) > 0)
+			return
```

Comment:
Might also want to check distance between `user` and `W`, so you can't pull something into the plater from the far side of it. Minor thing but good practice